force_bdss/utilities.py

```diff
@@ -51,6 +51,8 @@ def nested_getstate(state_dict):
         try:
             if isinstance(state_dict[key], (tuple, list)):
                 state_dict[key] = [el.__getstate__() for el in state_dict[key]]
+            elif isinstance(state_dict[key], dict):
```

Comment:
```suggestion
            elif not isinstance(state_dict[key], dict):
```
And remove lines 55 + 56?

Reply:
Thank you, this is a very good suggestions. To the best of my knowledge, these two approaches should be equivalent? If yes, I will try to think whether the old explicit case is better / worse from the readability point of view. I believe you are absolutely right that we can merge these three lines with one simple `not` statement; I am very bad however at correctly identifying the truthy statements with `not`.

I will look into this and come back to you asap, thank you.

Reply:
No worries, yes, I think they should be equivalent. 

I sometimes get confused with nested logic statements, so reducing the number of lines can help simplify things. However, I also agree that using `not` is more implicit, so please use whatever way you think is more clear

Reply:
I have a feeling that it is not the only type we should treat explicitly. However, I can't narrow down the exact situation when this will break. Something like `d = Dict(value_trait = List(CustomHasTraits))` would definitely require a custom `__getstate__`.

After talking to Mark today, it seems that having a generic `__getstate__` is not the best idea overall. We should not hesitate to implement object-specific gestate methods, if we want to maintain different traits versions compatibility.

Reply:
That is a very good point. I have been thinking along similar lines recently - or at least, feeling that we should probably leave the `__getstate__` super traits method alone and implement the serialization in a separate method.